Code/GraphMol/molopstest.cpp

```diff
@@ -1976,6 +1976,100 @@ void testAddHsCoords() {
   delete m;
   delete m2;
 
+  // make sure Hs are on the xy plane if conformer is 2D
+  smi = "C=C";
+  m = SmilesToMol(smi);
+  TEST_ASSERT(m);
+  TEST_ASSERT(m->getNumAtoms() == 2);
+  conf = new Conformer(2);
+  m->addConformer(conf);
+  conf->setAtomPos(0, RDGeom::Point3D(0, 0, 0));
+  conf->setAtomPos(1, RDGeom::Point3D(1.3, 0, 0));
+  conf->set3D(false);
+
+  m2 = MolOps::addHs(*m, false, true);
+
+  conf2 = &(m2->getConformer());
+
+  TEST_ASSERT(m2->getNumAtoms() == 6);
+
+  TEST_ASSERT(feq(conf2->getAtomPos(2).z, 0.0));
+  TEST_ASSERT(feq(conf2->getAtomPos(3).z, 0.0));
+  TEST_ASSERT(feq(conf2->getAtomPos(4).z, 0.0));
+  TEST_ASSERT(feq(conf2->getAtomPos(5).z, 0.0));
+  delete m;
+  delete m2;
+
+  // make sure NHs are on the same plane as the double bond
+  smi = "NC=C";
+  unsigned int nh2Idx = 0;
+  unsigned int chIdx = 1;
+  unsigned int ch2Idx = 2;
+  m = SmilesToMol(smi);
+  TEST_ASSERT(m);
+  TEST_ASSERT(m->getNumAtoms() == 3);
+  conf = new Conformer(3);
+  m->addConformer(conf);
+  conf->setAtomPos(nh2Idx, RDGeom::Point3D(1.759236, 0.825542, 1.347849));
+  conf->setAtomPos(chIdx, RDGeom::Point3D(0.817392, 0.181048, 2.180373));
+  conf->setAtomPos(ch2Idx, RDGeom::Point3D(-0.070943, 0.888262, 2.875625));
+
+  m2 = MolOps::addHs(*m, false, true);
+
+  conf2 = &(m2->getConformer());
+
+  TEST_ASSERT(m2->getNumAtoms() == 8);
+
+  auto nh2NbrRange = boost::make_iterator_range(
+      m2->getAtomNeighbors(m2->getAtomWithIdx(nh2Idx)));
+  std::vector<ROMol::vertex_descriptor> nh2His(2);
+  auto it = std::copy_if(nh2NbrRange.begin(), nh2NbrRange.end(), nh2His.begin(),
```

Comment:
In the line above you've restricted the size of `nh2His` to be 2, but then you use this general approach to find the H neighbors. That feels inconsistent.
Since you know the order in which the Hs are added (and we don't say anywhere that you shouldn't rely upon this), I think it makes things simpler and considerably more readable to just specifically use the atom IDs of the Hs in the testing code